src/convertToObject.js

```diff
@@ -6,7 +6,55 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  if (!sourceString.length) {
+    return {};
+  }
+
+  const rawLines = sourceString.split(';');
+
+  // filter lines without letters
+  const noEmptyLines = rawLines
+    .map((line) => {
+      return line.trim();
+    })
+    .filter((item) => item.length > 0);
+
+  const aggregatedLines = noEmptyLines.reduce((prev, line) => {
+    if (line.includes(':')) {
+      prev.push(line);
+    } else {
+      prev[prev.length - 1] += line;
+    }
+
+    return prev;
+  }, []);
+
+  // remove ';'
+  const noSemicolon = aggregatedLines.map((item) => {
+    // Array[string]
+    const [property, value] = item.split(':');
+    const newValue = value.split(';').join('');
+
+    return `${property}:${newValue}`;
+  });
+
+  // trim
+  const noSpaces = noSemicolon.map((item) => {
+    const [property, value] = item.split(':');
+
+    const newKey = property.trim();
+    const newValue = value.trim();
+
+    return `${newKey}:${newValue}`;
+  });
+
+  const convertToObj = noSpaces.reduce((prev, item) => {
+    const [key, value] = item.split(':');
+
+    return { ...prev, [key]: value };
+  }, {});
```

Comment:
[CODE PRACTICE] - Avoid using loops in this task, use iteration methods instead. The `reduce` method here is being used in a loop-like manner to construct an object. Consider using `Object.fromEntries` along with `map` to create the object in a more declarative way.